### ISSUE TITLE
Missing overloads of assertNotEquals for primitive types

### ISSUE THREAD

**Issue by HK201** [junit-team/junit5#1962]
Version: JUnit Jupiter 5.3.2

```java
@Test
void equalsVsNotEquals() {
  assertEquals(0, 0L);
  assertNotEquals(0, 0L);
}
```

Both of the above assertions will work. The reason is that `assertEquals` provides overloads for primitive types, while `assertNotEquals` doesn't. In other words:

- `assertEquals`: Actually compares the numbers -> 0 == 0
- `assertNotEquals`: Compares the types -> Int != Long

I'd expect `assertEquals` to be the opposite of `assertNotEquals`, which is not the case.

**Comment by sbrannen** [comment]
This was addressed in 5.4.

If you upgrade to 5.4 or higher (preferably 5.5.1), you'll see that the latter is invoked for the second assertion.

```java
	/**
	 * <em>Assert</em> that {@code expected} and {@code actual} are not equal.
	 *
	 * @since 5.4
	 */
	@API(status = STABLE, since = "5.4")
	public static void assertNotEquals(long unexpected, long actual) {
		AssertNotEquals.assertNotEquals(unexpected, actual);
	}
```

And that results in:

```
org.opentest4j.AssertionFailedError: expected: not equal but was: <0>
```

I am therefore closing this issue.

**Comment by sbrannen** [comment]
This is a duplicate of #1607.

Please search closes issues and PRs before opening a new issue.
